statsmodels/tsa/mstl.py

```diff
@@ -0,0 +1,240 @@
+"""
+Author: Kishan Manani
+License: BSD-3 Clause
+
+An implementation of MSTL [1], an algorithm for time series decomposition when
+there are multiple seasonal components.
+
+This implementation has the following differences with the original algorithm:
+- Missing data must be handled outside of this class.
+- The algorithm proposed in the paper handles a case when there is no
+seasonality. This implementation assumes that there is at least one seasonal
+component.
+
+[1] K. Bandura, R.J. Hyndman, and C. Bergmeir (2021)
+MSTL: A Seasonal-Trend Decomposition Algorithm for Time Series with Multiple
+Seasonal Patterns
+https://arxiv.org/pdf/2107.13462.pdf
+"""
+from typing import Dict, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import pandas as pd
+from scipy.stats import boxcox
+
+from statsmodels.tools.typing import ArrayLike1D
+from statsmodels.tsa.seasonal import STL
+from statsmodels.tsa.tsatools import freq_to_period
+
+
+class MSTL:
+    """
+    MSTL(endog, periods=None, windows=None, lmbda=None, iterate=2, **kwargs)
```

Comment:
I don't like **kwargs.  It would be better to use a specific parameter `stl_args` or `stl_kwargs` as a `dict[str,bool | int | None]`. The alternative would be to list each of the STL arguments as keyword only arguments, although there are a lot.